lib/round.rb

```diff
@@ -0,0 +1,78 @@
+require 'pry'
+require_relative'./deck'
+require_relative'./turn'
+require_relative './card'
+
+class Round
+    attr_reader :deck, :turns, :correct_answer
+    def initialize(deck)
+        @deck = deck
+        @turns = []
+    end
+
+    def current_card; @deck.cards.first end
+
+    def take_turn(guess)
+        @turns << Turn.new(guess, current_card)
+        @deck.cards = @deck.cards.rotate(1)
+        @turns.last
+    end
+
+    def number_correct
+        @correct = 0
+        @turns.each do |turn|
+            @correct += 1 if turn.correct? == true
+        end
+        @correct
+    end
+
+    def number_correct_by_category(input)
+        @correct_by_category = 0
+        @turns.each do |turn|
+            if turn.card.category == input
+                if turn.card.answer.capitalize == turn.guess.capitalize
+                    @correct_by_category += 1
+                end
+            end
+        end
+        @correct_by_category
+    end 
+
+    def percent_correct; (number_correct / @turns.count.to_f)*100 end
+
+    def percent_correct_by_category(input)
+        total_by_category = 0
+        @turns.each do |turn|
+            if turn.card.category == input
+                total_by_category += 1 
+            end
+        end
+        (number_correct_by_category(input) / total_by_category.to_f)*100
+    end
+
+    def start
+        puts "Welcome! You're playing with 4 cards. \n--------------------------------------"
```

Comment:
Even though when running your application with `ruby lib/runner.rb` there are 8 cards added to the deck, because this line is hard-coded, it always prints `"Welcome! You're playing with 4 cards."` to the console.

A better choice would have been to use some interpolation here that outputted the actual number of cards in the deck.

`"Welcome! You're playing with #{@deck.cards.count} cards.`